apps/cow-tools/src/app/milkman/(components)/SingleOrderForm/Twap.tsx

```diff
@@ -0,0 +1,95 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect } from "react";
+import { Controller, FieldValues, useForm } from "react-hook-form";
+
+import { Checkbox } from "#/components/Checkbox";
+import { Input } from "#/components/Input";
+import { Select, SelectItem } from "#/components/Select";
+import { Form } from "#/components/ui/form";
+import { orderTwapSchema } from "#/lib/schema";
+
+import { TransactionStatus } from "../../utils/type";
+import { FormFooter } from "./Footer";
+
+export enum TWAP_DELAY_OPTIONS {
+  TEN_MINUTES = "10 minutes",
+  THIRTY_MINUTES = "30 minutes",
+  ONE_HOUR = "1 hour",
+  THREE_HOURS = "3 hours",
+  TWELVE_HOURS = "12 hours",
+  ONE_DAY = "1 day",
+}
+
+export const TwapDelayValues = {
+  [TWAP_DELAY_OPTIONS.TEN_MINUTES]: 10 * 60,
+  [TWAP_DELAY_OPTIONS.THIRTY_MINUTES]: 30 * 60,
+  [TWAP_DELAY_OPTIONS.ONE_HOUR]: 60 * 60,
+  [TWAP_DELAY_OPTIONS.THREE_HOURS]: 3 * 60 * 60,
+  [TWAP_DELAY_OPTIONS.TWELVE_HOURS]: 12 * 60 * 60,
+  [TWAP_DELAY_OPTIONS.ONE_DAY]: 24 * 60 * 60,
+};
+
+export function FormTwap({
+  onSubmit,
+  defaultValues,
+}: {
+  onSubmit: (data: FieldValues) => void;
+  defaultValues?: FieldValues;
+}) {
+  const form = useForm<typeof orderTwapSchema._type>({
+    resolver: zodResolver(orderTwapSchema),
+  });
```

Comment:
what I mean in https://github.com/bleu-fi/monorepo/pull/493/files#r1407663244:
```suggestion
    resolver: zodResolver(orderTwapSchema),
    defaultValues
  });
```